Api key support and Energy consumption metric

mandinger opened the pull request:
I added a new query and filled a statistics entity with energy consumption data from the past 10 days, it keeps the 1h refresh cadence, so when new data is avaliable will be automatically pulled into the energy consumption.

mandinger committed:
feat: added energy consumption

mandinger committed:
Merge pull request #1 from mandinger/consumo

feat: added energy consumption

mandinger committed:
Added support for Api key authentication, new sensor for consumption. (#3)

* Update sensor.py

* added new login flow

mandinger committed:
feat: backfill energy usage

mandinger committed:
fix: error when importing a second time

mandinger committed:
feat: added backfill the month

mandinger committed:
feat: update first setup

First setup brings current month of consumption
Following refreshes only request missing and future data

mandinger committed:
fix: line breaks

b-per commented:
I just moved to Octopus and this PR seems to be what I am looking for!

I used to use https://github.com/ad-ha/repsolluzygas-async for tracking consumption and was looking for a similar option for Octopus ES. I'm still in the process of getting my provider switched so I don't have any consumption yet showing on the octopus website, but if need be I am happy to try this PR/branch when I get some any consumption showing up.

mandinger committed:
Fix breaking change on 2025.11

has_mean has been deprecated for mean_type

mandinger committed:
Added unit class and new unit class

Not passing this parameters has also being deprecated
https://developers.home-assistant.io/blog/2025/10/16/recorder-statistics-api-changes/

mandinger commented:
@b-per I noticed that my changes stopped working on the 2025-11 release
https://developers.home-assistant.io/blog/2025/10/16/recorder-statistics-api-changes/
I added a fix, maybe this will help you get the energy dashboard working

kallewesterling commented:
@MiguelAngelLV ¿hay alguna novedad sobre cuándo podrías hacer el merge de esta rama con tu paquete para facilitar la integración en Home Assistant?
